File: android/app/src/main/java/com/prince/turtlekeyboard/integration/SuggestedShortcut.java
package com.prince.turtlekeyboard.integration;

/**
 * One pre-baked shortcut a {@link SuggestedShortcutSource} offers when an app is
 * enrolled. The trigger {@code /name} shows up in the Quick Panel; tapping Go commits
 * the {@link #template} into the host editor.
 */
public final class SuggestedShortcut {

    public final String name;
    public final String label;
    public final String emoji;
    public final String template;
    public final boolean needsPrompt;

    public SuggestedShortcut(String name, String label, String emoji,
                             String template, boolean needsPrompt) {
        this.name = name;
        this.label = label;
        this.emoji = emoji;
        this.template = template;
        this.needsPrompt = needsPrompt;
    }
}

File: ios/TurtleKeyboardExtension/Integration/SuggestedShortcut.swift
import Foundation
#if os(iOS)
import UIKit

// MARK: - SuggestedShortcut
//
// Port of android/.../integration/SuggestedShortcut. Surfaces a curated
// palette of pre-filled templates above the keys when entering an empty
// field. Tap → either insert the template directly, or seed the command
// bar with it for the user to refine.

struct SuggestedShortcut {
    let name: String          // unique id, e.g. "followup"
    let label: String         // chip display label, e.g. "Follow-up"
    let emoji: String         // single-char badge, e.g. "📧"
    let template: String      // text inserted on tap (verbatim when needsPrompt = false)
    let needsPrompt: Bool     // true → seed the command bar; false → insert directly
}

// MARK: - FieldKind
//
// iOS keyboard extensions cannot read the host app's bundle id (the way
// Android keys the catalog off `EditorInfo.packageName`). We derive the
// closest signal we *can* see — field traits via `UITextDocumentProxy`
// + `UITextInputTraits` — and bucket fields into a coarse kind enum.

enum FieldKind {
    case sensitive    // password, OTP, CVV — never show shortcuts here
    case email
    case url
    case search
    case numeric
    case general

    static func from(_ input: InputContext) -> FieldKind {
        if input.looksSensitive { return .sensitive }

        switch input.textContentType {
        case .some(.emailAddress): return .email
        case .some(.URL):          return .url
        default: break
        }

        switch input.keyboardType {
        case .emailAddress:
            return .email
        case .URL:
            return .url
        case .webSearch:
            return .search
        case .numberPad, .decimalPad, .numbersAndPunctuation,
             .phonePad, .asciiCapableNumberPad:
            return .numeric
        default:
            return .general
        }
    }
}
#endif
